lib/ansible/executor/module_common.py

```diff
@@ -587,52 +587,67 @@ def _slurp(path):
+    interpreter_config = u'ansible_%s_interpreter' % interpreter_name
+    # key for config
     interpreter_config_key = "INTERPRETER_%s" % interpreter_name.upper()
 
-    if C.config.get_configuration_definitions().get(interpreter_config_key):
-        # a config def exists for this interpreter type; consult config for the value
-        interpreter_out = C.config.get_config_value(interpreter_config_key, variables=task_vars)
-        discovered_interpreter_config = u'discovered_interpreter_%s' % interpreter_name
+    interpreter_out = None
+    # skip detection for network os execution, use playbook supplied one if possible
+    if remote_is_local:
+        interpreter_out = task_vars.get('ansible_playbook_%s' % interpreter_name)
+
+    # a config def exists for this interpreter type; consult config for the value
+    elif C.config.get_configuration_definition(interpreter_config_key):
+
+        interpreter_from_config = C.config.get_config_value(interpreter_config_key, variables=task_vars)
+        interpreter_out = templar.template(interpreter_from_config.strip())
 
-        interpreter_out = templar.template(interpreter_out.strip())
+        # handle interpreter discovery if requested or empty interpreter was provided
+        if not interpreter_out or interpreter_out in ['auto', 'auto_legacy', 'auto_silent', 'auto_legacy_silent']:
 
-        facts_from_task_vars = task_vars.get('ansible_facts', {})
+            discovered_interpreter_config = u'discovered_interpreter_%s' % interpreter_name
+            facts_from_task_vars = task_vars.get('ansible_facts', {})
 
-        # handle interpreter discovery if requested
-        if interpreter_out in ['auto', 'auto_legacy', 'auto_silent', 'auto_legacy_silent']:
             if discovered_interpreter_config not in facts_from_task_vars:
                 # interpreter discovery is desired, but has not been run for this host
                 raise InterpreterDiscoveryRequiredError("interpreter discovery needed",
                                                         interpreter_name=interpreter_name,
                                                         discovery_mode=interpreter_out)
             else:
                 interpreter_out = facts_from_task_vars[discovered_interpreter_config]
-    else:
-        # a config def does not exist for this interpreter type; consult vars for a possible direct override
-        interpreter_config = u'ansible_%s_interpreter' % interpreter_name
-
-        if interpreter_config not in task_vars:
-            return None, interpreter
-
-        interpreter_out = templar.template(task_vars[interpreter_config].strip())
 
-    shebang = u'#!' + interpreter_out
-
-    if args:
-        shebang = shebang + u' ' + u' '.join(args)
+    # lacking config (which should include vars) consult vars for a possible direct override
+    elif interpreter_config in task_vars:
+        interpreter_out = templar.template(task_vars.get(interpreter_config).strip())
+    else:
```

Comment:
This is a hardcoded case that seems like it shouldn't exist.

Reply:
set python interpreter to '' (i had in ansible.cfg), i'll add test case